tests/scripts/all.sh

```diff
@@ -4344,10 +4344,9 @@ support_build_sha_armce() {
     clang_ver="$(clang --version|grep version|sed -E 's#.*version ([0-9]+).*#\1#')"
 
     # we need asm/hwcap.h available for runtime detection
-    echo '#include <asm/hwcap.h>' | clang -E - >/dev/null 2>&1
-    have_hwcap=$?
+    echo '#include <asm/hwcap.h>' | clang -E - >/dev/null 2>&1 && no_hwcap=0 || no_hwcap=1
 
-    [[ "${clang_ver}" -ge 4 && "${have_hwcap}" -eq 0 ]]
+    [[ "${clang_ver}" -ge 4 && "${no_hwcap}" -eq 0 ]]
```

Comment:
So rather than the double negative, how about a variable `have_hwcap` that gets set to `0` or `1` accordingly?